sources/packages/backend/libs/test-utils/src/factories/student-scholastic-standing.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  Application,
+  StudentAssessment,
+  StudentScholasticStanding,
+  StudentScholasticStandingChangeType,
+  User,
+} from "@sims/sims-db";
+
+/**
+ * Creates a scholastic standing record ready to be saved to the database.
+ * @param relations dependencies.
+ * - `submittedBy` institution user submitting the scholastic standing change.
+ * - `application` related student application.
+ * - `studentAssessment` related assessment.
+ * - `student` related student.
+ * @returns scholastic standing record ready to be saved to the database.
+ */
+export function createFakeStudentScholasticStanding(relations?: {
+  submittedBy: User;
+  application?: Application;
+  studentAssessment?: StudentAssessment;
+}): StudentScholasticStanding {
+  const scholasticStanding = new StudentScholasticStanding();
+  scholasticStanding.application = relations?.application;
+  scholasticStanding.submittedData = {} as unknown;
+  scholasticStanding.submittedDate = new Date();
+  scholasticStanding.submittedBy = relations?.submittedBy;
+  scholasticStanding.note = null;
+  scholasticStanding.studentAssessment = relations?.studentAssessment;
+  scholasticStanding.referenceOffering = null;
+  scholasticStanding.unsuccessfulWeeks = null;
+  scholasticStanding.changeType =
```

Comment:
if we want a scholastic standing with different `change type`, is the plan make this as a generic function or  will be creating scholastic standing for each change type

Reply:
For any `createFakeSomething` I believe that the idea is to use as below.
```ts
const scholasticStanding = createFakeStudentScholasticStanding();
scholasticStanding.changeType = StudentScholasticStandingChangeType.StudentWithdrewFromProgram;
scholasticStanding.myOtherProperty = "My other value";
```
When should we create a parameter? When the parameter will do something else than a direct assignment.

Reply:
oh..okay, that means there will be a "create something" with some default value, and it can we overwritten in each tests as per the requirement